werkzeug/security.py

```diff
@@ -86,22 +73,23 @@ def pbkdf2_bin(data, salt, iterations=DEFAULT_PBKDF2_ITERATIONS,
                      string name of a known hash function or a function
                      from the hashlib module.  Defaults to sha256.
     """
-    if isinstance(hashfunc, string_types):
-        hashfunc = _hash_funcs[hashfunc]
-    elif not hashfunc:
-        hashfunc = hashlib.sha256
+    if not hashfunc:
+        hashfunc = 'sha256'
+
     data = to_bytes(data)
     salt = to_bytes(salt)
 
     # If we're on Python with pbkdf2_hmac we can try to use it for
     # compatible digests.
     if _has_native_pbkdf2:
```

Comment:
Fairly sure all supported Python's have native pbkdf2, and the `if hash_name` check below will always be `True` as written, so this entire function can be simplified.

Reply:
Might have to look into this more. Under what circumstances is `hashfunc` callable but doesn't have a name?

Reply:
Good to know. I've cleaned it up.

Reply:
@davidism actually I don't know. I just copied it from the old code. I believe it will always has a `name`.